File: Infrastructure-Module/persistence/src/main/java/com/bamyanggang/persistence/experience/mapper/TagMapper.java
package com.bamyanggang.persistence.experience.mapper;

import com.bamyanggang.domainmodule.domain.experience.aggregate.Tag;
import com.bamyanggang.persistence.experience.jpa.entity.TagJpaEntity;
import org.springframework.stereotype.Component;

@Component
public class TagMapper {
    public TagJpaEntity toJpaEntity(Tag tag) {
        return TagJpaEntity.of(tag.getId(), tag.getName(), tag.getParentTagId(), tag.getUserId());
    }

    public Tag toDomainEntity(TagJpaEntity tagJpaEntity) {
        return Tag.Companion.toDomain(tagJpaEntity.getId(),
                tagJpaEntity.getName(),
                tagJpaEntity.getParentTagId(),
                tagJpaEntity.getUserId()
        );
    }
}

File: Infrastructure-Module/persistence/src/main/java/com/bamyanggang/persistence/experience/mapper/TagMapper.kt
package com.bamyanggang.persistence.experience.mapper

import com.bamyanggang.domainmodule.domain.experience.aggregate.Tag
import com.bamyanggang.persistence.experience.jpa.entity.TagJpaEntity
import org.springframework.stereotype.Component

@Component
class TagMapper {
    fun toJpaEntity(tag: Tag): TagJpaEntity = TagJpaEntity.of(tag.id, tag.name, tag.parentTagId, tag.userId)

    fun toDomain(parentTagJpaEntity: TagJpaEntity): Tag =
        Tag.toDomain(parentTagJpaEntity.id,
            parentTagJpaEntity.name,
            parentTagJpaEntity.parentTagId,
            parentTagJpaEntity.userId
        )
}